.github/workflows/build.yml

```
@@ -18,39 +18,34 @@ jobs:
       matrix:
         id:
           - "mysql-8.0"
-          - "mysql-5.7"
     steps:
       - name: Parse ID
         id: parse
         run: |
           set -x
+          if [ "${GITHUB_REF_TYPE}" = "tag" ] && [[ "${GITHUB_REF_NAME}" =~ ^${{ matrix.id }}\. ]]; then
+            version="${GITHUB_REF_NAME}"
+          else
+            version=${{ matrix.id }}-latest
+          fi
+
           push=false
-          case "${{ github.ref }}" in
-            refs/tags/*)
-              version=$(echo "${{ github.ref }}" | sed -e "s,^refs/tags/,,g")
-              case "${version}" in
-                ${{ matrix.id }}.*)
-                  push=true
-                  ;;
-                *)
-                  version=${{ matrix.id }}-latest
-                  push=false
-                  ;;
-              esac
-              ;;
-            *)
-              version=${{ matrix.id }}-latest
-              push=true
-              ;;
-          esac
+          if [ "${GITHUB_EVENT_NAME}" = "push" ]; then
+            push=true
+            if [ "${GITHUB_REF_TYPE}" = "tag" ] && [[ !("${GITHUB_REF_NAME}" =~ ^${{ matrix.id }}\.) ]]; then
```

bashで正規表現を何回も使うのはしんどいのでこんなんでいけないかな。

```suggestion
            if [ "${GITHUB_REF_TYPE}" = "tag" ] && [ "${version}" != "${GITHUB_REF_NAME}” ]; then
```

---

タシカニ。`$version` を見る発想はあったんですが、`-latest` の有無で判断すべきという思考から抜け出せなかったです…。